Cron/Providers/ProcessedOldNotificationsProvider.php

```diff
@@ -0,0 +1,63 @@
+<?php
+/**
+ *
+ * Adyen Payment module (https://www.adyen.com/)
+ *
+ * Copyright (c) 2024 Adyen N.V. (https://www.adyen.com/)
+ * See LICENSE.txt for license details.
+ *
+ * Author: Adyen <magento@adyen.com>
+ */
+
+namespace Adyen\Payment\Cron\Providers;
+
+use Adyen\Payment\Api\Repository\AdyenNotificationRepositoryInterface;
+use Adyen\Payment\Helper\Config;
+use Adyen\Payment\Logger\AdyenLogger;
+use Magento\Framework\Api\SearchCriteriaBuilder;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Store\Model\StoreManagerInterface;
+
+class ProcessedOldNotificationsProvider implements NotificationsProviderInterface
+{
+    public function __construct(
+        private readonly AdyenNotificationRepositoryInterface $adyenNotificationRepository,
+        private readonly SearchCriteriaBuilder $searchCriteriaBuilder,
+        private readonly Config $configHelper,
+        private readonly StoreManagerInterface $storeManager,
+        private readonly AdyenLogger $adyenLogger
+    ) { }
+
+    public function provide(): array
+    {
+        $storeId = $this->storeManager->getStore()->getId();
+        $numberOfDays = $this->configHelper->getRequiredDaysForOldWebhooks($storeId);
+
+        $dateFrom = date('Y-m-d H:i:s', time() - $numberOfDays * 24 * 60 * 60);
+
+        $searchCriteria = $this->searchCriteriaBuilder
+            ->addFilter('done', 1)
+            ->addFilter('processing', 0)
+            ->addFilter('created_at', $dateFrom, 'lteq')
+            ->create();
+
+        try {
+            $items = $this->adyenNotificationRepository->getList($searchCriteria);
+            return $items->getItems();
+        } catch (LocalizedException $e) {
+            $errorMessage = sprintf(
+                __('An error occurred while providing notifications older than %s days!'),
```

Comment:
Can we add the store name or ID here? Maybe that's useful to see in the log line to be able to debug

Reply:
Please see the related commit [here](https://github.com/Adyen/adyen-magento2/pull/2837/commits/e8d5fd1552725689a59127bae20b0b7cee71e350)